duckduckgo/iOS
# Etag OOS fix

Task/Issue URL: 
- https://app.asana.com/0/856498667320406/1128900793643895
- https://app.asana.com/0/856498667320406/1109742356053775

**Description**:
- Fixes a problem of missing content blocker list despite existing etag. 
- Handling weak etags, by ignoring the 'W/' prefix.
- Do not store etags for http bloom filter requests (as they use spec instead for update processing).
- Introduce statistic for configuration fetches.

**Steps to test this PR**:

Regular scenario
1. Comment out `try disconnectMeStore.persist(data: data)` from `StorageCache`.
2. Run the app (fresh install) - list should be downloaded but not stored. Content blocker should not be available.
3. Restart app and check etag storage logs - no etag for disconnect me list should be returned.

Migration scenario
1. Check out develop.
2. Comment out `try? disconnectMeStore.persist(data: data)` from `StorageCache`.
3. Run the app (fresh install) - list should be downloaded but not stored. Content blocker should not be available.
4. Restart app and check etag storage logs for disconnectMe list query - entry should be present (this is bug).
5. Checkout this branch (and clear changes).
6. Run the app - disconnectMe should be fetched and stored despite matching etag. 
7. Ensure content blocking is working.

Weak etags
1. Run the app (fresh install), see that `httpsWhitelist` is downloaded (with Weak etag) and stored.
2. Restart the app - see in logs that `httpsWhitelist` etag entry in etag storage has no weak prefix.
3. Observe that `httpsWhitelist` was not updated after restart.

Config Fetch info
- Run the app -> on startup info Pixel should be fired.
- In case of background download, pixel is not fired, but info is incremented.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
